mono/mini/tramp-amd64.c

```diff
@@ -360,7 +341,7 @@ mono_arch_create_generic_trampoline (MonoTrampolineType tramp_type, MonoTrampInf
 			g_assert (r11_save_code == after_r11_save_code);
 
 			/* Copy from the save slot into the register array slot */
-			amd64_mov_reg_membase (code, i, AMD64_RSP, r11_save_offset + orig_rsp_to_rbp_offset, sizeof (target_mgreg_t));
+			amd64_mov_reg_membase (code, i, AMD64_RSP, r11_save_offset + orig_rsp_to_rbp_offset + framesize, sizeof (target_mgreg_t));
 			amd64_mov_membase_reg (code, AMD64_RBP, saved_regs_offset + (i * sizeof (target_mgreg_t)), i, sizeof (target_mgreg_t));
```

Comment:
Is this still needed ?

Reply:
Not by this PR, but this PR didn't add support to save R11 and the above change is just a fix of existing logic since original implementation didn't correctly restore it, so original implementation didn't work as intended (https://github.com/mono/mono/commit/68c2d43e65580467de1f708413d79178f40a3258). Question is if the initial reason trying to save and restore R11 still holds, if so I believe we should keep storing/restoring R11.